geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitor.java

```diff
@@ -25,76 +26,63 @@
 /**
  * This class acts as a monitor and listen for VM stats update on behalf of MemberMBean.
  *
- *
  */
 public class VMStatsMonitor extends MBeanStatsMonitor {
-
-  private static final int VALUE_NOT_AVAILABLE = -1;
-
-  private volatile float cpuUsage = 0;
-
-  private static String processCPUTimeAttr = "ProcessCpuTime";
-
+  static final float VALUE_NOT_AVAILABLE = -1;
+  private static final String PROCESS_CPU_TIME_ATTRIBUTE = "ProcessCpuTime";
   private long lastSystemTime = 0;
-
   private long lastProcessCpuTime = 0;
+  private final boolean processCPUTimeAvailable;
+  private AtomicInteger cpuUsageBits = new AtomicInteger(Float.floatToIntBits(0f));
 
-  private boolean processCPUTimeAvailable;
+  public float getCpuUsage() {
+    return Float.intBitsToFloat(cpuUsageBits.get());
+  }
 
   public VMStatsMonitor(String name) {
     super(name);
-    processCPUTimeAvailable = MBeanJMXAdapter.isAttributeAvailable(processCPUTimeAttr,
+    processCPUTimeAvailable = MBeanJMXAdapter.isAttributeAvailable(PROCESS_CPU_TIME_ATTRIBUTE,
         ManagementFactory.OPERATING_SYSTEM_MXBEAN_NAME);
     if (!processCPUTimeAvailable) {
-      cpuUsage = VALUE_NOT_AVAILABLE;
+      cpuUsageBits.set(Float.floatToIntBits(VALUE_NOT_AVAILABLE));
     }
-
   }
 
-
-  @Override
-  public void handleNotification(StatisticsNotification notification) {
-
-    for (StatisticId statId : notification) {
-      StatisticDescriptor descriptor = statId.getStatisticDescriptor();
-      String name = descriptor.getName();
-      Number value;
-      try {
-        value = notification.getValue(statId);
-      } catch (StatisticNotFoundException e) {
-        value = 0;
-      }
-      log(name, value);
-      statsMap.put(name, value);
-    }
-    refreshStats();
+  long currentTimeMillis() {
+    return System.currentTimeMillis();
```

Comment:
We should update this class to not use System.currentTimeMillis. We are trying to compute elapsed time and System.currentTimeMillis has problems with this when the machine's wall clock time is adjusted.
Use System.nanoTime (and convert to millis if needed) or use the newer "Instant" and "Duration" classes.
See: https://www.baeldung.com/java-measure-elapsed-time

Reply:
Sure, I recall seeing an specific JIRA for something similar but didn't remember it when doing the changes.
Found it!: https://issues.apache.org/jira/browse/GEODE-5563.